### ISSUE TITLE
Nice job! Here are my questions about the loss functions in Section 3.4 Counterfactual-Faithful Training.

### ISSUE THREAD

**Issue by shyern** [yue-zhongqi/gcm-cf#2]
Nice job! Here are my questions about the loss functions in Section 3.4 Counterfactual-Faithful Training.
(1) As I can see, there are three loss functions, i.e., Eq.(6), Eq.(7), and Eq.(8) in the main paper. But the contrastive loss in Eq.(7) and the WGAN loss in Eq.(8) are actually used during the training of "cf.py". Can you explain why the loss function in Eq.(6) is not used in the training procedure?
(2) There are a serise of loss functions for disentangle in "train_tfvaegan_inductive.py", e.g., "zy_disentangle, ys_disentangle, yx_disentangle, ...". But I can't find any explanation in the main paper or usage in the code about these disentangle loss functions. Can you explain the meaning of these disentangle losses and how to use them?

**Comment by yue-zhongqi** [comment]
Hi,

Sorry for the late reply due to recent deadlines.
(1) beta-VAE loss is used during training. Please refer to https://github.com/yue-zhongqi/gcm-cf/blob/bb38adc7a17020ae0761395feaa891bec24e33f9/zsl/train_tfvaegan_inductive.py#L436
(2) Those were experimented but not used to obtain the results in the main paper.

Thanks & regards,
Zhongqi
